### ISSUE TITLE
Unexpected with_fileglob behavior change from 1.1 to 1.2

### ISSUE THREAD

**Issue by lorin** [ansible/ansible#2749]
I just noticed a change in the behavior of with_fileglob in devel that broke one of my playbooks. Here's a simple playbook to demonstrate the issue

```

---
- hosts: all
  tasks:
    - debug: msg=$item
      with_fileglob:
        - ../keys/*
```

If I run this under devel (f16751de292594c), I see that `item` only contains the file name, no path:

```
TASK: [debug msg=$item] *******************************************************
ok: [vagrant] => (item=vagrant.pub) => {"item": "vagrant.pub", "msg": "vagrant.pub"}
```

If I run this under v1.1, `item`  includes the relative path:

```
TASK: [debug msg=$item] *********************
ok: [vagrant] => (item=../keys/vagrant.pub) => {"item": "../keys/vagrant.pub", "msg": "../keys/vagrant.pub"}
```

This broke one of my tasks that looked like this:

```
- name: add ssh public keys
  authorized_key: user=admin key="$FILE($item)"
  with_fileglob: ../keys/public/admin/*
```

**Comment by mpdehaan** [comment]
I would agree this feels broken.

Investigating.

**Comment by mpdehaan** [comment]
Fix pushed in bd0c228 ... let me know if this works for you.

I simplified some other things in the module previously while doing some code cleanup, and it's clear that went a little too far.

Thanks for the heads up.

**Comment by lorin** [comment]
Yup, this works for me now. Thanks, Michael.
